tests/judges/test_instruction_judge.py

```diff
@@ -0,0 +1,298 @@
+        
+        judge = InstructionJudge(
+            name='Error Register Judge', user_instructions='This registration will fail'
+        )
+
+        result = judge.register_scorer()
+
+        # Should return None and log warning
+        self.assertIsNone(result)
+        mock_logger.warning.assert_called_once()
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_optimize_with_sufficient_data(self, mock_make_judge):
+        """Test optimize method with sufficient traces."""
+        mock_judge = Mock()
+        mock_aligned_judge = Mock()
+        mock_judge.align.return_value = mock_aligned_judge
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(name='Optimize Test Judge', user_instructions='Test optimization')
+
+        # Create mock traces (need at least 10 for MIN_EXAMPLES_FOR_OPTIMIZATION)
+        traces = [Mock() for _ in range(12)]
+
+        result = judge.optimize(traces)
+
+        # Verify alignment was called and judge was updated
+        mock_judge.align.assert_called_once_with(traces=traces)
+        self.assertEqual(judge.scorer_func, mock_aligned_judge)
+        self.assertTrue(result)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    @patch('server.judges.instruction_judge.logger')
+    def test_register_scorer_failure_returns_none(self, mock_logger, mock_make_judge):
+        """Test register_scorer returns None on failure."""
+        mock_judge = Mock()
+        mock_judge.register.side_effect = Exception("Registration failed")
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Registration Fail Judge', user_instructions='Test registration failure'
+        )
+
+        result = judge.register_scorer()
+
+        # Should return None and log warning
+        self.assertIsNone(result)
+        mock_logger.warning.assert_called_once()
+
+    @patch('server.judges.instruction_judge.make_judge')
+    @patch('server.judges.instruction_judge.logger')
+    def test_optimize_alignment_failure(self, mock_logger, mock_make_judge):
+        """Test optimize method when alignment fails."""
+        mock_judge = Mock()
+        mock_judge.align.side_effect = Exception("Alignment failed")
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(name='Alignment Fail Judge', user_instructions='Test alignment failure')
+
+        traces = [Mock() for _ in range(12)]  # Sufficient traces
+        result = judge.optimize(traces)
+
+        # Should return False and log error
+        self.assertFalse(result)
+        mock_logger.error.assert_called_once()
+        self.assertIn("Alignment failed", mock_logger.error.call_args[0][0])
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_scorer_function_creation(self, mock_make_judge):
+        """Test that _create_scorer returns the MLflow judge directly."""
```

Comment:
Generally not a big fan of testing private methods. Why is it important to test this? Is this an implementation detail?